tests/helpers/test_service.py

```diff
@@ -226,181 +224,192 @@ def area_mock(hass):
     )
 
 
-class TestServiceHelpers(unittest.TestCase):
-    """Test the Home Assistant service helpers."""
+async def test_service_call(hass: HomeAssistant):
+    """Test service call with templating."""
+    calls = async_mock_service(hass, "test_domain", "test_service")
+    config = {
+        "service": "{{ 'test_domain.test_service' }}",
+        "entity_id": "hello.world",
+        "data": {
+            "hello": "{{ 'goodbye' }}",
+            "effect": {"value": "{{ 'complex' }}", "simple": "simple"},
+        },
+        "data_template": {"list": ["{{ 'list' }}", "2"]},
+        "target": {"area_id": "test-area-id", "entity_id": "will.be_overridden"},
+    }
 
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        self.calls = mock_service(self.hass, "test_domain", "test_service")
+    await hass.async_add_executor_job(service.call_from_config, hass, config)
```

Comment:
These tests are the only place in the code base were `service.call_from_config` is called.

Should the tests be rewritten to call `service.async_call_from_config` instead, and we just add one test which calls the sync version?

Reply:
Tests updated to use `service.async_call_from_config` directly